elastic/kibana
# [Fleet] Fix Step 1 in policy editor not loading when agent policy already contains a limited integration

## Summary

Fixes #110924. This PR fixes the UX when a selected agent policy already has a limited package (that is, a package that can only be added _once_ per agent policy).

To test:
1) Add Endpoint package to any policy
2) From that policy's details page (inside Fleet > Agent policies), click `Add integration`
3) Browse to Endpoint package again and click to add it
4) Previously, the first step would be blank and the second step would not have any policy selected. Observe now that the first step loads and the second step selects the policy the user entered from, but displays an error, and the Save button is disabled

![image](https://user-images.githubusercontent.com/1965714/135942917-eccbcf21-14b9-42b9-9796-5f861eb7fca8.png)

Conversely, if the user enters from Integrations > Endpoint Security and _all_ policies already have Endpoint package, a different error message is shown (in this case the user can create an new policy inline):

![image](https://user-images.githubusercontent.com/1965714/135943696-87dfabcf-49a6-4588-8747-dbf70a7ff107.png)

## Additional notes

The changes here also incidentally fixes #112485 due to re-rendering triggers being corrected by this PR (namely, the package information is now being retrieved at the parent `CreatePackagePolicyPage` component level, instead of inside `StepSelectAgentPolicy`). To test that bug, use the Synthetics package which has a custom policy editor UI.

I also removed some legacy code intended to support a previous flow where a package was allowed to be selected in-line within the policy editor.
